doc/design/cluster_train/submit-job.md

```diff
@@ -0,0 +1,133 @@
+      |-quick_start
+        |-trainer.py
+        |-dataset.py
+        |-requirements.txt
+  ```
+
+## Submit Distributed Training Job With Python Code
+<img src="./src/submit-job-python.png" width="800">
+
+- `paddle.job.dist_train()` will call the Job Server API `/v1/packages` to upload the trainer package and save them on CephFS, and then call `/v1/trainer/job` to submit the PaddlePaddle distributed job.
+- `/v1/trainer/job` will start a building job for preparing the runtime Docker image. When the building job is finished, Job Server will submit the PaddlePaddle distributed job to Kubernetes.
+- *NOTE*: For the first version, we will not prepare the runtime docker image, instead, we will mount the trainer package in a temporary folder into the base docker image. We will not support custom Python dependencies in the first version as well.
+
+You can call `paddle.job.dist_train` and provide distributed training configuration as the parameters:
+```python
+paddle.job.dist_train(
+    trainer=paddle.trainer.SGD(...,
+                              paddle.updater.Adam(...)),
+    reader=reader,
+    paddle_job=PaddleJob(
+      pserver_bucket="stander",
+      base_image="yancey1989/paddle-cloud",
+      job_name="paddle-job",
+      namespace="paddle-cloud",
+      cpu_num=3,
+      memory="1G"
+      trainer_package="/example/word2vec",
+      entry_point="python %s" % __file__))
+```
+
+The pseudo code of `paddle.job.dist_train` is as follows:
+```python
+  def dist_train(trainer, reader, num_passes=1, event_handler=None, feeding=None, paddle_job=None):
+    # if the code is running on cloud, set PADDLE_ON_CLOUD=YES
+    if os.getenv("PADDLE_ON_CLOUD", "NO") == "NO":
+      #submit the paddle job
+      paddle_job.submit()
+    else:
+      #start the training
+      trainer.train(reader, num_passes, event_handler, feeding)
+```
+
+parameter | required | default | explain
+  --- | --- | --- | ---
+job_name|YES||you should special a unique job name which in a namespace
+entry_point|YES|| entry point for startup trainer process
+trainer_package|YES|| trainer package file path, you can special a cloud path with `pfs://home/paddle` or a normal path with `/home/paddle`
+base_image|YES||PaddlePaddle production Docker image
+memory|YES|| memory for the trainers totally used
+cpu_num|YES|1| CPU count for the trainers totally used
+gpu_num|NO|0| GPU count for the trainers totally used
+pserver_bucket|NO|mini| you can special `pserver-bucket` for the PServer resource
+pservers|NO|1| custom PServer count
+pserver_cpu|NO|1| custom PServer CPU count
+pserver_memory|NO|1| custom PServer memory limit
+
+### Special Resource for a Distributed Training Job
+- PServer Resource
+  - Special `pserver_bucket`
+    - `pserver_bucket=mini`, a single PServer process, it's suitable for learning how to Paddle Cloud.
+    - `pserver_bueckt=stander`, many PServer processes.
+    - `pserver_bucket=premium`, large PServer processes
+  - Custom PServer Resource
```

Comment:
Done, the same as https://github.com/PaddlePaddle/Paddle/pull/1770#discussion_r115931796